src/wallet/wallet.h

```diff
@@ -805,7 +805,7 @@ class CWallet final : public CCryptoKeyStore, private interfaces::Chain::Notific
     /**
      * populate vCoins with vector of available COutputs.
      */
-    void AvailableCoins(interfaces::Chain::Lock& locked_chain, std::vector<COutput>& vCoins, bool fOnlySafe=true, const CCoinControl *coinControl = nullptr, const CAmount& nMinimumAmount = 1, const CAmount& nMaximumAmount = MAX_MONEY, const CAmount& nMinimumSumAmount = MAX_MONEY, const uint64_t nMaximumCount = 0, const int nMinDepth = 0, const int nMaxDepth = 9999999) const EXCLUSIVE_LOCKS_REQUIRED(cs_wallet);
```

Comment:
> It's ugly if code calling this with nMinDepth (eg, other PRs or out-of-tree patches) can magically have it become nMaxDepth instead.

agreed. 

> Maybe nMaxDepth ought to be moved to CCoinControl also? 

was thinking the same. any suggestions on how to go about answering that? or should I just go for it and pull `nMaxDepth` out? 

> Or removed? Is there a use case for this?

looks like the `listunspent` rpc accepts a `nMaxDepth` & passes that through [here](https://github.com/bitcoin/bitcoin/pull/15906/files#diff-df7d84ff2f53fcb2a0dc15a3a51e55ceR2799)

Reply:
>     It's ugly if code calling this with nMinDepth (eg, other PRs or out-of-tree patches) can magically have it become nMaxDepth instead.

Could solve this by swapping the arguments `bool fOnlySafe` and `CCoinControl` of the function, so that all out-of-tree patches have a (silent) merge conflict and won't compile anymore?

Alternative is to also pass nMaxDepth through with the coincontrol.

Reply:
pulled onto coincontrol